rotkehlchen/api/rest.py

```diff
@@ -5409,3 +5409,62 @@ def force_refetch_evm_transactions(
             transaction_count += (after_count - before_count)
 
         return _wrap_in_ok_result({'new_transactions_count': transaction_count})
+
+    def addresses_interacted_before(
+            self,
+            from_address: ChecksumEvmAddress,
+            to_address: ChecksumEvmAddress,
+    ) -> Response:
+        with self.rotkehlchen.data.db.conn.read_ctx() as cursor:
+            cursor.execute(
+                'SELECT COUNT(*) FROM history_events JOIN evm_events_info ON '
+                'history_events.identifier=evm_events_info.identifier WHERE '
+                'location_label=? AND address=?',
+                (from_address, to_address),
+            )
+            return api_response(_wrap_in_ok_result(result=cursor.fetchone()[0] > 0))
+
+    @async_api_call()
+    def prepare_token_transfer(
+            self,
+            from_address: ChecksumEvmAddress,
+            to_address: ChecksumEvmAddress,
+            token: EvmToken,
+            amount: FVal,
+    ) -> dict[str, Any]:
+        manager = self.rotkehlchen.chains_aggregator.get_chain_manager(
+            blockchain=token.chain_id.to_blockchain(),  # type: ignore  # checked in the validation
```

Comment:
if it is checked in the validation, you can just use that as a parameter here. something like:
```py
def prepare_token_transfer(
            self,
            blockchain: SUPPORTED_EVM_CHAINS_TYPE,
            from_address: ChecksumEvmAddress,
            to_address: ChecksumEvmAddress,
            token: EvmToken,
            amount: FVal,
    ) -> dict[str, Any]:
```